File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
  let ans = 0,
      minPrice = Number.MAX_SAFE_INTEGER;
  
  for(let price of prices) {
    minPrice = Math.min(minPrice, price);
    ans = Math.max(ans, price - minPrice);
  }
  
  return ans;
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = 100001, N = prices.length, ans = 0;
        
        for(int i = 0; i < N; i++) {
            min = Math.min(min, prices[i]);
            ans = Math.max(ans, prices[i] - min);
        }
        
        return ans;
    }
}